src/stores/walletStore/walletStore.test.tsx

```diff
@@ -301,4 +327,93 @@ describe('walletStore', () => {
       }),
     );
   });
+  test('should save wallet type to localStorage when connecting', async () => {
+    announceProvider(metamask);
+
+    await walletStore.connectWallet({
+      walletProvider: WalletProvider.EIP6963,
+      providerId: 'metamask-uuid',
+    });
+
+    expect(localStorageMock.setItem).toHaveBeenCalledWith(
+      'walletType',
+      'MetaMask',
+    );
+  });
+
+  test('should clear wallet type in localStorage when disconnecting', async () => {
+    announceProvider(metamask);
+
+    await walletStore.connectWallet({
+      walletProvider: WalletProvider.EIP6963,
+      providerId: 'metamask-uuid',
+    });
+
+    //  Clear mock call history after connect
+    localStorageMock.setItem.mockClear();
```

Comment:
This is kind of useful if you've already checked the first call and will only need the most recent one later on, rather than `localStorageMock.setItem.mock.calls[localStorageMock.setItem.mock.calls.length - 1]`